workspaces/optic-engine/src/queries/endpoint.rs

```diff
@@ -41,6 +41,32 @@ impl<'a> EndpointQueries<'a> {
   }
 
   pub fn resolve_path(&self, path: &str) -> Option<PathComponentIdRef> {
+    // If the path is a root path, we need to check it separately
+    // There is always going to be a node for the root path, so we need to check
+    // If it has been learnt, there will be an associated HTTP method
+    if path.eq("/") {
```

Comment:
I think this is correct, we have to treat this as a special case since we always have a root node, so resolving the path isn't enough to check if we have "seen" this path. I think we need to treat this if it has _any_ http method associated with this.

I'm not sure if I've missed anything here though

Reply:
Hmm, I think resolve_path should just be concerned with paths. perhaps the logic might be more clear if we used petgraph's DFS/BFS visitors. However, I believe the problem is:
```
let mut path_components = path.split('/');
    // skip leading empty
    path_components.next();
    let mut last_resolved_path_id = Some(ROOT_PATH_ID);
    while let Some(s) = path_components.next() {
      // eprintln!("trying to match segment {}", s);
      let node_index = self
        .graph_get_index(last_resolved_path_id.unwrap())
        .expect("expected a node with node_id to exist");
```

So when the path is `/`, we split by `'` skip the first `''` and try to search for the second `''` relative to the root path component. IIRC at the time this was written I didn't want to make assumptions about component names being empty. So we should be able to say if path.eq("/") return Some(ROOT_PATH_ID). 

With any change in behavior we should see a change in tests, so whatever the solution ends up being, I'd like to see a test that would fail with the previous version of the code that would pass with this one. Sorry for the bug :)

Reply:
I added two tests here to illustrate (also, not sure if we should move these into modules + have separate tests):
- `resolve_path_can_resolves_root_path` - before this PR, the code always returned `None` where it should have returned root
- `resolve_path_does_not_include_root_when_no_methods_attached` - implementing `if path.eq("/") return Some(ROOT_PATH_ID)` _always_ returns the root node - even if we haven't learnt anything. Given that we always have the root node, we treat the root node as being seen if we've seen any HTTPmethods learnt from this. Otherwise what happens is we always see the root node as learnt - even if we haven't learnt any endpoints
<img width="480" alt="Screen Shot 2021-06-18 at 3 38 45 PM" src="https://user-images.githubusercontent.com/18374483/122621886-c3bcff80-d04b-11eb-806c-9e8056f2c821.png">

This seemed incorrect to me, which is why there's extra logic in there - but perhaps the logic isn't in the right place, but I'm not sure where a better place is to put this. Thoughts?

Reply:
Well definitely resolve_path should only care about paths. I think it's ok that / shows up as a known path, but i might be missing what else that actually affects?

Reply:
I think that is the only difference, they both show up in the undocumentedUrls component and go through the same flows. I've updated this to remove the extra logic such that it's just `if path.eq("/") return Some(ROOT_PATH_ID)`

Reply:
Great. I think that makes the most sense; there kind of has to be an established root path. Curious the bug made it this long haha.